### PR TITLE
🎨 Split sign in and sign up url

### PR DESCRIPTION
**Pull request by williamchong007**

### CODE REVIEW

File: src/components/SlidingMenu.vue [hunk 1]
```diff
@@ -106,7 +106,7 @@ export default {
     ]),
   },
   methods: {
-    getOAuthLoginAPI,
+    getOAuthRegisterAPI,
```
Comment (nwingt) [inline src/components/SlidingMenu.vue]: use computed instead?

File: src/server/api/routes/users.js [hunk 2]
```diff
@@ -33,10 +33,16 @@ router.get('/users/self', async (req, res, next) => {
   }
 });
 
+router.get('/users/register', (req, res) => {
+  const state = crypto.randomBytes(8).toString('hex');
+  req.session.state = state;
+  res.redirect(getOAuthURL({ state, isLogin: false }));
+});
+
```
Comment (nnkken) [inline src/server/api/routes/users.js]: How about this?

```JS
const loginOrRegisterRoute = ({ isLogin }) => (req, res) => {
  const state = crypto.randomBytes(8).toString('hex');
  req.session.state = state;
  res.redirect(getOAuthURL({ state, isLogin }));
};

router.get('/users/register', loginOrRegisterRoute({ isLogin: false }));
router.get('/users/login', loginOrRegisterRoute({ isLogin: true }));
```

Comment (nwingt) [reply src/server/api/routes/users.js]: How about this
```javascript
function newSessionState(req, res, next) {
  req.session.state = crypto.randomBytes(8).toString('hex');
  next()
}

...

router.get('/users/register', newSessionState, (req, res) => {
  res.redirect(getOAuthURL({ state: req.session.state, isLogin: false }));
});

router.get('/users/login', newSessionState, (req, res) => {
  res.redirect(getOAuthURL({ state: req.session.state, isLogin: true }));
});
```

Comment (williamchong007) [reply src/server/api/routes/users.js]:
```
function setSessionOAuthState(req) {
  const state = crypto.randomBytes(8).toString('hex');
  req.session.state = state;
}



router.get('/users/register', (req, res) => {
  setSessionOAuthState(req);
  res.redirect(getOAuthURL({ state: req.session.state, isLogin: false }));
});

```
?

Comment (nwingt) [reply src/server/api/routes/users.js]: I mean using middleware

**Pull request merged**